Webpack build failed when generating monolith with skipUserManagement

<!--
- Please follow the issue template below for bug reports and feature requests.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug or feature request, if an error is being thrown a stack trace helps -->
webpack build failed when generating monolith with skipUserManagement
```
ERROR in ./src/main/webapp/app/core/index.ts
Module not found: Error: Can't resolve './user/user.service' in '/Users/gongy/poc/workflow-app/app/src/main/webapp/app/core'
 @ ./src/main/webapp/app/core/index.ts 15:21-51
 @ ./src/main/webapp/app/app.module.ts
 @ ./src/main/webapp/app/app.main.ts
```
mvn
```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/gongy/poc/workflow-app/app/src/test/java/net/topikachu/caseapp/security/DomainUserDetailsServiceIntTest.java:[4,36] cannot find symbol
  symbol:   class User
  location: package net.topikachu.caseapp.domain
```
##### **Motivation for or Use Case**

<!-- Explain why this is a bug or a new feature for you -->
I'd like to generate a monolith with an external IAM system. No user managment is required in this new app.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->
`jhipster --skip-user-management`
Choose monolith with other stuffs

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

<!-- For bug reports, if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->
It seems there's a wrong config in the
webapp/app/core/index.ts

Also the DomainUserDetailsServiceIntTest should be removed from the generated app.
##### **JHipster Version(s)**

<!--
Which version of JHipster are you using, is it a regression?
-->
generator-jhipster@5.0.0-beta.1

##### **JHipster configuration**
Using JHipster version installed locally in current project's node_modules
Executing jhipster:info
Options: 
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
app@0.0.0 /Users/gongy/poc/workflow-app/app
└── generator-jhipster@5.0.0-beta.1 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "net.topikachu.caseapp",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.0-beta.1",
    "baseName": "app",
    "packageName": "net.topikachu.caseapp",
    "packageFolder": "net/topikachu/caseapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "yarn",
    "applicationType": "monolith",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "zh-cn"
    ],
    "skipUserManagement": true
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)

git version 2.10.1

node: v8.11.1

npm: 5.6.0

yarn: 1.3.2

Docker version 18.05.0-ce-rc1, build 33f00ce

docker-compose version 1.21.0, build 5920eb0
<!--
To provide all information we need, you should run `jhipster info` in the project root folder, and
copy/paste the result here.
The `.yo-rc.json` file generated in the root folder is mandatory for bug reports. This will help us to replicate the scenario.
You should remove any sensitive information like the rememberMe key or the jwtSecretKey key.
-->

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<!--
If the error is during an entity creation or associated with a specific entity.
If you are using JDL, please share that configuration as well.
-->

##### **Browsers and Operating System**

<!-- What OS are you on? is this a problem with all browsers or only IE8? -->
macOS Sierra
- [ X] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/client/templates/angular/src/main/webapp/app/core/index.ts.ejs
+++ b/generators/client/templates/angular/src/main/webapp/app/core/index.ts.ejs
@@ -24,5 +24,7 @@ export * from './auth/user-route-access-service';
 export * from './login/login-modal.service';
 <%_ } _%>
 export * from './login/login.service';
+<%_ if (!skipUserManagement || authenticationType === 'oauth2') { _%>
 export * from './user/user.service';
+<%_ } _%>
 export * from './core.module';

--- a/generators/server/files.js
+++ b/generators/server/files.js
@@ -925,7 +925,7 @@ const serverFiles = {
             ]
         },
         {
-            condition: generator => generator.applicationType === 'monolith' && generator.authenticationType !== 'oauth2',
+            condition: generator => !shouldSkipUserManagement(generator) && generator.authenticationType !== 'oauth2',
             path: SERVER_TEST_SRC_DIR,
             templates: [
                 // Create auth config test files

